MacrobenchmarkSample/baseBenchmarks/src/main/java/com/example/benchmark/macro/base/scroll/ScrollBenchmark.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.benchmark.macro.base.scroll
+
+import androidx.benchmark.macro.BaselineProfileMode
+import androidx.benchmark.macro.CompilationMode
+import androidx.benchmark.macro.FrameTimingMetric
+import androidx.benchmark.macro.StartupMode
+import androidx.benchmark.macro.StartupTimingMetric
+import androidx.benchmark.macro.junit4.MacrobenchmarkRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.SdkSuppress
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.Direction
+import com.example.benchmark.macro.base.util.DEFAULT_ITERATIONS
+import com.example.benchmark.macro.base.util.TARGET_PACKAGE
+import junit.framework.TestCase.fail
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * Scrolls down and up in the default activity.
+ */
+@RunWith(AndroidJUnit4::class)
+class ScrollBenchmark {
+
+    @get:Rule
+    val benchmarkRule = MacrobenchmarkRule()
+
+    @Test
+    @SdkSuppress(minSdkVersion = 24)
+    fun noCompilation() = scroll(CompilationMode.None())
+
+    @Test
+    fun defaultCompilation() = scroll(CompilationMode.DEFAULT)
+
+    @Test
+    @SdkSuppress(minSdkVersion = 24)
+    fun partialBaselineProfileDisabled() =
+        scroll(
+            CompilationMode.Partial(
+                baselineProfileMode = BaselineProfileMode.Disable,
+                warmupIterations = 3
+            )
+        )
+
+    @Test
+    @SdkSuppress(minSdkVersion = 24)
+    fun partialBaselineProfileRequired() =
+        scroll(
+            CompilationMode.Partial(
+                baselineProfileMode = BaselineProfileMode.Require,
+                warmupIterations = 3
+            )
+        )
+
+    @Test
+    fun full() = scroll(CompilationMode.Full())
+
+    private fun scroll(compilationMode: CompilationMode) =
+        benchmarkRule.measureRepeated(
+            packageName = TARGET_PACKAGE,
+            metrics = listOf(StartupTimingMetric(), FrameTimingMetric()),
+            compilationMode = compilationMode,
+            startupMode = StartupMode.COLD,
+            iterations = DEFAULT_ITERATIONS
+        ) {
+            startActivityAndWait()
+            val scrollableObject = device.findObject(By.scrollable(true))
```

Comment:
how about waiting for some scrollable object as well?
```kotlin
device.wait(Until.hasObject(By.scrollable(true)), 5_000)
```
This way, if an app has something loading asynchronously (which hopefully it would), the test wouldn't crash and be happy.